File: mParticle-Apple-SDK/AppEnvironmentProvider.m
#import "AppEnvironmentProvider.h"

@implementation AppEnvironmentProvider

- (BOOL)isAppExtension {
#if TARGET_OS_IOS == 1
    return [[NSBundle mainBundle].bundlePath hasSuffix:@".appex"];
#else
    return NO;
#endif
}
@end

File: mParticle-Apple-SDK-Swift/Sources/Utils/AppEnvironmentProvider.swift
import Foundation

@objc public protocol AppEnvironmentProviderProtocol {
    func isAppExtension() -> Bool
}

@objc(AppEnvironmentProvider) public final class AppEnvironmentProvider: NSObject, AppEnvironmentProviderProtocol {
    private let bundlePath: String

    @objc public init(bundlePath: String) {
        self.bundlePath = bundlePath
        super.init()
    }

    @objc public override convenience init() {
        self.init(bundlePath: Bundle.main.bundlePath)
    }

    @objc public func isAppExtension() -> Bool {
        #if os(iOS)
        return bundlePath.hasSuffix(".appex")
        #else
        return false
        #endif
    }
}
